src/krate/mod.rs

```diff
@@ -65,6 +65,7 @@ pub struct Crate {
     pub homepage: Option<String>,
     pub documentation: Option<String>,
     pub readme: Option<String>,
+    pub readme_file: Option<String>,
```

Comment:
I see that we're now storing the README filename alongside its content, though it doesn't seem to be used anywhere ? I'm not opposed to storing it but I'm not sure to see how useful it is to store it in the database.

Reply:
You're right, we don't strictly need to store it. I was thinking we might need it in `render-readmes`, but that pulls the README filename from the manifest directly. However, I feel like storing this as well might be important for things like #1025/#1019?

A more general argument is that, if there's no need to store the README filename in the database, then there's no need to store the README _content_ in the database, because we can't properly make sense of the latter without the former.

That's moreorless the motivation of this PR in a nutshell: we don't know how to interpret the README contents without knowing its filename. So if we store the content, we should store the filename too.

Reply:
Yeah, you're totally right about needing to store the filename alongside the README content, my bad.

However, maybe we can improve a little here and instead of storing the README filename as transmitted, how about storing the normalized README type ? i.e only `markdown` for now. This would also allow us to use an `ENUM` instead of a `VARCHAR` field in the database and simplify its usage.

Reply:
I like the idea of doing this, but I'm worried about the loss of fidelity. e.g. say we do so, and currently store `'markdown'` or `NULL` for everything. Later we add AsciiDoc or org-mode support — it'd be great to go back and rerender the newly supported readmes, but to do so we'd have to reach back inside each crate and retrieve the readme filename from the manifest again.

I think storing the filename directly allows us more flexibility later on in how we want to treat certain things, especially in the face of changing/new standards and formats; this is how we approach file classification at GitHub, for example, and it works well for us.

Reply:
> but to do so we'd have to reach back inside each crate and retrieve the readme filename from the manifest again.

In any case, we probably will have to do it. The `crates` table only stores the README of the last published version of a given crate, so if we add org-mode or AsciiDoc support to crates.io we'd still have to reach back inside each crate in order to render it. Anyway, you're right about your approach being more flexible so let's keep it this way :)